ape_frame/accounts.py

```diff
@@ -34,25 +43,63 @@ def alias(self) -> str:
     def address(self) -> AddressType:
         return self.web3.eth.accounts[0]
 
-    def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
-        try:
-            raw_signature = self.web3.eth.sign(self.address, hexstr=_hash_eip191_message(msg).hex())
-        except ValueError as e:
-            if not e.args[0]["message"] == "User declined transaction":
-                raise
+    def sign_message(self, msg: Any, **signer_options: Any) -> Optional[MessageSignature]:
+        raw_signature = None
 
-            return None
+        if isinstance(msg, str):
+            # encode string messages as Ethereum Signed Messages
+            try:
+                raw_signature = self.web3.eth.sign(self.address, text=msg)
+            except ValueError as e:
+                if not e.args[0]["message"] == "User declined transaction":
```

Comment:
problem: not every `ValueError` has args with values in there. You can raise an empty ValueError, which will cause an `IndexError` here which is not what we want.

nit: reverse the condition

one more nit: Add a comment about why this excepted.

```python
if e.args and e.args[0]["message"] != "User declined transaction":
    raise  # Is an unknown ValueError

# Return None so Ape handles the decline properly.
return None
```

Reply:
it seems like this is a problem all over